### PR TITLE
Add support for Apache Flink

### PR DESCRIPTION
**Pull request by rmoff**
In the latest version of Apache Flink (1.18) [a JDBC driver
was added](https://cwiki.apache.org/confluence/display/FLINK/FLIP-293%3A+Introduce+Flink+Jdbc+Driver+For+Sql+Gateway) for Flink's SQL Gateway. 

This PR adds direct support for this driver to Dbeaver.

### CODE REVIEW

**Comment by LonwoLonwo** [comment]
Thanks for your contribution.

If you have a logo, you can also add it like this:

```xml
icon="icons/flink_icon.png"
iconBig="icons/flink_icon_big.png"
logoImage="icons/flink_logo.png"
```

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]:
```suggestion
                    <file type="jar" path="maven:/org.apache.flink:flink-sql-jdbc-driver-bundle:RELEASE"/>
```

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]:
```suggestion
                    <file type="license" path="https://www.apache.org/licenses/LICENSE-2.0.txt"/>
```

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]:
```suggestion
```

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]: We do not have the Flink driver for now, so there is no need for this code. We will add it in the future if it will be the reason.

**Comment by rmoff** [comment]
Hi @LonwoLonwo, I already have `icon` and `iconBig` defined. None of the other generic providers use `logoImage` and looking at the code it default to using `icon` so I think this should be ok? 

icon shows up correctly in the UI: 

image

CleanShot 2023-11-14 at 09 28 07

image

**Comment by LonwoLonwo** [comment]
Ok, no problem. It can be just a big icon on the Connection Page.

**Comment by LonwoLonwo** [comment]
And please take a look at my review comments.

**Comment by rmoff** [comment]
> And please take a look at my review comments.

Hi, I don't see any review comments?

**Review by LonwoLonwo — CHANGES_REQUESTED** [review]

**Comment by LonwoLonwo** [comment]
> Hi, I don't see any review comments?

Sorry, they were probably stuck somehow.

**Review by LonwoLonwo — APPROVED** [review]

**Comment by LonwoLonwo** [comment]
Ok. Now we are waiting for one more review and then the QA Team.

**Comment by rmoff** [comment]
Great, thank you @LonwoLonwo !

**Review by E1izabeth — APPROVED** [review]

**Comment by rmoff** [comment]
@LonwoLonwo do I need to do anything else here? What's the usual leadtime on the next step? thanks!

**Comment by LonwoLonwo** [comment]
We are waiting for a test database instance from our dev-ops team now. Also, we have big release now, so, it all can be not so fast.

**Comment by LonwoLonwo** [comment]
Hello @rmoff 

I checked your PR with a test environment.

And I see here:

```
java.lang.UnsupportedOperationException
	at org.apache.flink.table.jdbc.FlinkDatabaseMetaData.getTables(FlinkDatabaseMetaData.java:139)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCDatabaseMetaDataImpl.getTables(JDBCDatabaseMetaDataImpl.java:1116)
	at org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel.prepareTableLoadStatement(GenericMetaModel.java:586)
	at org.jkiss.dbeaver.ext.generic.model.TableCache.prepareLookupStatement(TableCache.java:66)
	at org.jkiss.dbeaver.ext.generic.model.TableCache.prepareLookupStatement(TableCache.java:1)
	at org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCStructLookupCache.reloadObject(JDBCStructLookupCache.java:113)
```

So I can't see my tables. And I can't create a table via UI. 
The only I can do is `SELECT 1`.

So, what is the workflow for this database and DBeaver? Why do we want to add a new driver?

**Comment by LonwoLonwo** [comment]
Hello again @rmoff 

I hope you are doing well.

Please take a look at my questions: https://github.com/dbeaver/dbeaver/pull/21842#issuecomment-1855694962.

**Comment by LonwoLonwo** [comment]
Ok. I'm closing this PR. 
For now. 
You can create a new one when you are ready.

**Pull request closed without merging**
